### ISSUE TITLE
grpo时，kl系数为时不时出现nan

### ISSUE THREAD

**Issue by evyy1234** [modelscope/ms-swift#4465]
日志如下：
{"loss": -0.03745745, "grad_norm": 0.00613374, "learning_rate": 4.6e-07, "memory(GiB)": 146.13, "train_speed(iter/s)": 0.004094, "completions/mean_length": 1047.175, "completions/min_length": 441.0, "completions/max_length": 1757.4, "completions/clipped_ratio": 0.175, "rewards/ToolAccuracy/mean": 0.775, "rewards/ToolAccuracy/std": 0.30299634, "reward": 1.55, "reward_std": 0.21547005, "kl": NaN, "clip_ratio/low_mean": NaN, "clip_ratio/low_min": NaN, "clip_ratio/high_mean": NaN, "clip_ratio/high_max": NaN, "clip_ratio/region_mean": NaN, "epoch": 0.46260601, "global_step/max_steps": "300/1296", "percentage": "23.15%", "elapsed_time": "20h 20m 38s", "remaining_time": "2d 19h 32m 31s"}

shell如下：
CUDA_VISIBLE_DEVICES=1,2  NPROC_PER_NODE=1 swift rlhf  --rlhf_type grpo  --model  /opt/db/qwen3-14b  --train_type lora --use_vllm true --dataset /opt/db/t2.jsonl    --max_completion_length 1289 --num_train_epochs 2 --learning_rate 5e-7 --gradient_accumulation_steps 2 --eval_steps 50 --save_steps 50  --temperature 0.9 --output_dir output  --warmup_ratio 0.05  --dataloader_num_workers 4 --dataset_num_proc 4  --num_generations 4  --log_completions true  --external_plugins  /opt/db/grpo/rew.py  --per_device_train_batch_size 4  --per_device_eval_batch_size 4 --reward_funcs  tool_acc  --reward_weights  2.0  --vllm_gpu_memory_utilization 0.97  --vllm_max_model_len 29000  --multi_turn_func tool_multi_turn --deepspeed zero3 --epsilon_high 0.28 --dynamic_sample true --overlong_filter true --max_resample_times 3 --deepspeed zero3    --vllm_enforce_eager  true

**Comment by hjh0119** [comment]
Because all the completions have been filtered out due to overlong
